pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -157,6 +157,21 @@ func (r *ReconcileInfinispan) Reconcile(request reconcile.Request) (reconcile.Re
 	// Apply defaults if not already set
 	infinispan.ApplyDefaults()
 
+	// Perform all the possible preliminary checks before go on
+	recResult, err := infinispan.PreliminaryChecks()
+	if err != nil {
+		if infinispan.SetCondition("preliminaryChecksFailed", "true", err.Error()) {
+			err1 := r.client.Status().Update(context.TODO(), infinispan)
+			reqLogger.Error(err1, "Could not update error conditions")
+		}
+		return *recResult, err
+	} else {
+		if infinispan.RemoveCondition("PreliminaryChecksFailed") {
```

Comment:
@rigazilla , looks like this PreliminaryChecksFailed need to start from lowercase also or not?

Reply:
oh yes!